app/src/main/java/com/venuenext/venuenextsdkdemo/MainActivity.kt

```diff
@@ -162,18 +190,37 @@ class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemS
     }
 
     private fun completeInitialize() {
+
         registerFCM()
+        VNOrderUI.initialize()
+        VNWalletUI.initialize()
+        VNPayment.configurePaymentProcessing(BraintreePaymentProcessableFragment())
+
+        VenueNext.notificationTitle = "Intel Demo"
+        VenueNext.notificationSmallIcon = R.mipmap.ic_launcher
+        VenueNext.notificationLargeIcon = R.mipmap.ic_launcher
+
+        VenueNext.configureAnalytics(FirebaseAnalytics(this@MainActivity))
+        VenueNext.configureAnalytics(LocalyticsAnalytics(this@MainActivity, application!!))
 
         bottomNavigationView.visibility = View.VISIBLE
+        navigateAfterInitialization()
 
+        // Only handle deep links after the SDK has loaded
+        handleIntent()
+    }
+
+    private fun completeInitializeAfterFailure(e: Throwable) {
+        Log.e(TAG, "SDK initialization failed!")
+        navigateAfterInitialization()
```

Comment:
(Optional) If the SDK initialization fails, can we even navigate? Just food for thought, I know we are doing this literally everywhere but maybe something to think about for the future.

Reply:
No, probably not. I’ll remove this.